modules/diskq/qdisk.c

```diff
@@ -351,13 +352,20 @@ qdisk_push_tail(QDisk *self, GString *record)
       return FALSE;
     }
 
-  if (!pwrite_strict(self->fd, (gchar *) &record_length, sizeof(record_length), self->hdr->write_head) ||
-      !pwrite_strict(self->fd, record->str, record->len, self->hdr->write_head + sizeof(record_length)))
+  ScratchBuffersMarker marker;
+  GString *msg_buffer;
+  msg_buffer = scratch_buffers_alloc_and_mark(&marker);
+
+  g_string_append_len(msg_buffer, ((gchar *) &record_length), sizeof(record_length));
```

Comment:
Reducing the number of disk writes (especially for a single length field) is a good direction, however we should do it without an extra move of the whole serialized message.
i.e. serialize it into the right buffer in the first place.

Discussed the topic with @alltilla IRL.
This commit is really good as a POC for reducing the number of writes, and he has a WIP code to eliminate the extra copy. I will leave this PR as is.